test/cppsuite/test_harness/connection_manager.h

```diff
@@ -99,6 +99,12 @@ class connection_manager {
         return session;
     }
 
+    WT_CONNECTION *
+    get_connection() const
```

Comment:
Instead of giving the connection to the timestamp manager we should wrap `set_timestamp` in the connection_manager so all connection API calls can be wrapped in the `_conn_mutex`.